### PR TITLE
fix(update): timeout increased for watcher [EE-6580]

### PR DESCRIPTION
**Pull request by prabhat83**
[EE-6580]

[EE-6580]: https://portainer.atlassian.net/browse/EE-6580?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: kubernetes/update.go [hunk 1]
```diff
@@ -141,7 +141,21 @@ func updateDeployment(ctx context.Context, deployCli v1.DeploymentInterface, dep
...
+		}
+
+		log.Warn().Err(err).Int("number of retries", numRetries).Msg("Deployment not ready yet, retrying")
+		numRetries++
+	}
+	if numRetries == 10 {
+		return errors.New("Something went wrong, deployment not ready after 10 retries")
+	}
+
+	return nil
```
Comment (prabhat83) [inline kubernetes/update.go]: Here I need to return an error

File: kubernetes/update.go [hunk 2]
```diff
@@ -141,7 +141,19 @@ func updateDeployment(ctx context.Context, deployCli v1.DeploymentInterface, dep
 		Str("deploymentName", deploymentName).
 		Msg("Waiting for deployment to complete")
 
-	return waitForDeployment(ctx, deployCli, newDeployment.Name, newDeployment.UID)
+	err = nil
+	numRetries := 1
+	for numRetries < 10 {
```
Comment (hookenz) [inline kubernetes/update.go]: Don't hard code 10.  Use const.

File: kubernetes/update.go [hunk 3]
```diff
@@ -141,7 +141,19 @@ func updateDeployment(ctx context.Context, deployCli v1.DeploymentInterface, dep
...
+		err = waitForDeployment(ctx, deployCli, newDeployment.Name, newDeployment.UID)
```
Comment (hookenz) [inline kubernetes/update.go]: Do you really need to retry the waitForDeployment or just increase the timeout?  It's a watch afterall and will return as soon as the deployment has completed.  We should give up after 5 minutes.   I don't think a retry makes sense personally unless you couldn't create the watch.

I'd make the timeout a parameter to `waitForDeployment`.  I don't see the point of re-creating a watch process personally.

There is also a 5s delay in the waitForDeployment function.   What's that about?  Is it sufficient?

File: kubernetes/update.go [hunk 4]
```diff
@@ -141,7 +141,19 @@ func updateDeployment(ctx context.Context, deployCli v1.DeploymentInterface, dep
...
+		if err == nil {
+			return nil
+		}
+
+		log.Warn().Err(err).Int("number of retries", numRetries).Msg("Deployment not ready yet, retrying")
+		numRetries++
+	}
+
+	return errors.WithMessage(err, "Something went wrong, deployment not ready after 10 retries")
```
Comment (hookenz) [inline kubernetes/update.go]: "Something went wrong" is not a very good error message.  

Also "github.com/pkg/errors"  should not be used.

**Review by hookenz — CHANGES_REQUESTED** [review]
See comments.  I don't agree with the retry loop here.
But also the error message could be better.  Discuss with me if you disagree.

File: kubernetes/update.go (hunk 3 above)
Comment (prabhat83) [reply kubernetes/update.go]: Got your point. Reverted the code

**Review by hookenz — APPROVED** [review]

**Review by hookenz — APPROVED** [review]

**Pull request merged**